components/profile-controller/api/v1beta1/profile_types.go

```diff
@@ -14,41 +13,51 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package v1alpha1
+package v1beta1
 
 import (
+	"k8s.io/api/core/v1"
 	rbacv1 "k8s.io/api/rbac/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
 )
 
+// Plugin is for customize actions on different platform.
+type Plugin struct {
+	metav1.TypeMeta `json:",inline"`
+	Spec            *runtime.RawExtension `json:"spec,omitempty"`
+}
+
+type ProfileCondition struct {
+	Type    string `json:"type,omitempty"`
+	Status  string `json:"status,omitempty" description:"status of the condition, one of True, False, Unknown"`
+	Message string `json:"message,omitempty"`
+}
```

Comment:
@jlewi @kunmingg should we add all the standard fields here or leave it as is for now?